plonky2x/src/frontend/eth/beacon/builder.rs

```diff
@@ -217,7 +217,7 @@ impl<F: RichField + Extendable<D>, const D: usize> CircuitBuilder<F, D> {
             withdrawals,
             idx,
         );
-        self.add_simple_generator(&generator);
+        self.add_simple_generator(generator.clone());
```

Comment:
Why did you change all of this?

Reply:
You shouldn't use the generator after you've added it